### ISSUE TITLE
CI actions are duplicated for pull requests

### ISSUE THREAD

**Issue by tymokvo** [pollination/queenbee#299]
The current GitHub actions settings are doubling the CI work by running on the `push` and `pull_request` events. See [here](https://github.com/pollination/queenbee/pull/295/checks) for an example. The runs on the left nav bar are identical.

One solution is to limit the `push` event to the master branch.

**Comment by mostaphaRoudsari** [comment]
Good point! Since we only merge the PR if they pass the tests we don't need to re-run them for the push! We should fix this in several repositories.
